### PR TITLE
YLP-2609 Patient "Target & alerts" view for HCP

### PR DESCRIPTION
**Pull request by celia-rouis**
## Checklist
- [x] PR name contains the Jira ticket number
- [x] Changelog updated
- [x] Code tested (unit tests and integration tests)
- [x] No console errors/warnings in integration tests
- [x] No translation files updated (except for English files)
- [x] Code formatted on every file
- [x] Copyright added on every file
- [x] End-to-end tests successful on CI

### CODE REVIEW

File: packages/yourloops/pages/patient-view/target-and-alerts/target-and-alerts-view.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+  const [saveInProgress, setSaveInProgress] = useState<boolean>(false)
+
+  useEffect(() => {
+    if (hash === '') {
+      return
+    }
+
+    const sectionId = hash.replace('#', '')
+    const section = document.getElementById(sectionId)
```
Comment (TimCanu) [inline packages/yourloops/pages/patient-view/target-and-alerts/target-and-alerts-view.tsx:69]: It is not advised in react to access/manipulate the dom using `document` (preferred way is to use refs).
More info: [here](https://www.shecodes.io/athena/43520-how-to-use-getelementbyid-with-react#:~:text=getElementById%20cannot%20be%20used%20directly,with%20the%20element%20that%20way.) and [here](https://stackoverflow.com/questions/72591035/can-i-use-the-document-object-in-react#:~:text=You%20can%2C%20but%20this%20goes,that%20is%20cross%2Dbrowser%20compatible.)

I don't think that it is much of an issue here (if I understood everything correctly), so I think that it is ok to keep as is if you don't feel like using refs.

File: packages/yourloops/pages/patient-view/target-and-alerts/target-and-alerts-view.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      console.error(error)
+      alert.error(t('patient-update-error'))
+      setSaveInProgress(false)
+    }
+  }
+
+  return (
+    <Container data-testid="target-and-alerts-container">
+      <Card variant="outlined" sx={{ padding: theme.spacing(2) }}>
+        <CardHeader title={t('target-and-alerts')} />
```
Comment (TimCanu) [inline packages/yourloops/pages/patient-view/target-and-alerts/target-and-alerts-view.tsx:92]: I felt that this tab is feeling weird/unfinished/drafty and one thing that could help is to have the divider as in the design. Did you forget it?
Screenshot from 2023-10-09 11-22-23

File: packages/yourloops/test/integration/pages/patient-view/dashboard/dashboard-view.hcp.spec.tsx [hunk 3]
```diff
@@ -165,11 +158,14 @@ describe('Patient dashboard for HCP', () => {
   })
 
   it('should be able to manage monitoring alerts parameters', async () => {
+    // Mocking the `scrollIntoView` method as it is not implemented in jsdom
+    Element.prototype.scrollIntoView = jest.fn()
```
Comment (TimCanu) [inline packages/yourloops/test/integration/pages/patient-view/dashboard/dashboard-view.hcp.spec.tsx:162]: Maybe this should be done in `packages/yourloops/test/integration/setup.js`?

File: packages/yourloops/test/integration/pages/patient-view/target-and-alerts/target-and-alerts-view.hcp.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+describe('Target and alerts view for HCP', () => {
+  const firstName = 'HCP firstName'
+  const lastName = 'HCP lastName'
+
+  const targetAndAlertsRoute = `${AppUserRoute.Patient}/${patient1Id}${AppUserRoute.TargetAndAlerts}`
+  const patientTargetAndAlertsRouteMmoL = `${AppUserRoute.Patient}/${patientWithMmolId}${AppUserRoute.TargetAndAlerts}`
+
+  beforeEach(() => {
+    mockWindowResizer()
```
Comment (TimCanu) [inline packages/yourloops/test/integration/pages/patient-view/target-and-alerts/target-and-alerts-view.hcp.spec.tsx:56]: Is this needed?

File: packages/yourloops/components/header-bars/patient-nav-bar-tabs.tsx [hunk 5]
```diff
@@ -37,12 +37,14 @@ import { makeStyles } from 'tss-react/mui'
 import { type Theme } from '@mui/material/styles'
 import Button from '@mui/material/Button'
 import GetAppIcon from '@mui/icons-material/GetApp'
-import { ChartTypes } from '../../enum/chart-type.enum'
+import { PatientView } from '../../enum/patient-view.enum'
 import PhonelinkSetupOutlinedIcon from '@mui/icons-material/PhonelinkSetupOutlined'
+import { Notifications } from '@mui/icons-material'
```
[the remark this answers is not in the document]
Comment (celia-rouis) [reply packages/yourloops/components/header-bars/patient-nav-bar-tabs.tsx:42]: Yes please!

File: packages/yourloops/pages/patient-view/target-and-alerts/target-and-alerts-view.tsx (hunk 2 above)
Comment (celia-rouis) [reply packages/yourloops/pages/patient-view/target-and-alerts/target-and-alerts-view.tsx:94]: I did 😬 Thanks for the sharp eyes

File: packages/yourloops/pages/patient-view/target-and-alerts/target-and-alerts-view.tsx (hunk 1 above)
Comment (celia-rouis) [reply packages/yourloops/pages/patient-view/target-and-alerts/target-and-alerts-view.tsx]: I replaced it and used a ref

**Comment by TimCanu** [comment]
Screenshot from 2023-10-09 16-40-25
What does the cancel button do?

**Pull request merged**